lib/linguist/heuristics.yml

```diff
@@ -714,6 +714,24 @@ disambiguations:
   - language: Markdown
     # Markdown syntax for scdoc
     pattern: '^#+\s+(NAME|SYNOPSIS|DESCRIPTION)'
+- extensions: ['.scm']
+  rules:
+  - language: Scheme
+    pattern:
+    - '(?:''[\(\w\*]|[\w\-]+->[\w\-]+|\b\.\.\.\b|\([+\-#:<>\/=~\)]|~>)'
+    - '^#:\w+'
+    - '#\w*\('
+    - '^\s*\((?i:define|import|library|let)'
```

Comment:
This is vulnerable to ReDoS attack. Please fix it, and ensure the new regex runs linearly and is a Re2-style regex. See https://github.com/github-linguist/linguist/pull/7242 for other regexes we're slowly cleaning up.

Reply:
You can find the magic [here](https://github.com/github-linguist/linguist/blob/1647b5be1658fb15a5e09bf89067f79b392301e2/lib/linguist/heuristics.rb#L69-L95).

My finding was based purely on testing the first regex in isolation 😉

Reply:
So I'm assuming comments aren't ignored from the match 🤔 ?